c2/c2-client-bundle/c2-client-service/src/main/java/org/apache/nifi/c2/client/service/C2HeartbeatFactory.java

```diff
@@ -225,4 +242,14 @@ private File getConfDirectory() {
 
         return confDirectory;
     }
+
+    public String calculateManifestHash(List<Bundle> loadedBundles) {
+        byte[] bytes = messageDigest.digest(loadedBundles.stream()
```

Comment:
Although this specific use of `MessageDigest.digest(byte[] bytes)` should be safe, other reuse of the `MessageDigest` instance is not safe. It would be better to construct a new instance of `MessageDigest` for each invocation of `calculateManifestHash`.

Reply:
Thanks, updated